packages/x-date-pickers/src/internals/components/PickersPopper.tsx

```diff
@@ -323,15 +324,17 @@ export const PickersPopper = (props: PickerPopperProps) => {
               ownerState={{ ...ownerState, placement }}
               {...otherPaperProps}
             >
-              {children}
-              <ActionBar
-                onAccept={onAccept}
-                onClear={onClear}
-                onCancel={onCancel}
-                onSetToday={onSetToday}
-                actions={[]}
-                {...componentsProps?.actionBar}
-              />
+              <PaperContent>
```

Comment:
As far as I've checked, I believe that only the `ActionBar` supports `componentProps` as it can accept changes in two ways:
- which actions to show via `componentProps`
- what to render via `ActionBar`

In this case, like you mentioned, we only allow user adding extra content before or after the main calendar content.

Reply:
It's a convention used by both the core and the data grid. When I can customize a slot with `components` I can also pass props with `componentsProps`. Most of the other slots are icons, and to be consistent with the data grid they should be supported too in `componentsProps`

Reply:
> As far as I've checked, I believe that only the ActionBar supports componentProps as it can accept changes in two ways:

Indeed, and that's something we should improve.
The pickers slots are not very mature yet.
But always having the props is a nice thing.
We don't pass the props to some icons because it's useless with the MUI icons for instance, and we have someone complaining because he uses custom icon components on which it would be usefull.

Reply:
I've added the `paperContent` props as suggested.
Also added a usage [example of it here](https://github.com/mui/mui-x/pull/5801/files#diff-198e8a43ff0e8f34f29cab37e8e3b3f57163ac164f0b9c9a813be213e2ea13b8R201).